### DISCUSSION TITLE
query regarding node_modules pushed

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [cockpit-project/cockpit-machines#1363] [General]
Can someone throw a light on why are node_modules pushed to the repo? Am i missing out on something or is it a mistake? I would've opened an issue directly but unsure whether that is a mistake or not.

**Comment by martinpitt** [comment]
It's deliberate, see https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#updating-node_modules . We keep node_modules/ in git to retain a history, guarantee reproducibility, make builds/checkouts much faster, and most of all: `npm install` should never ever run on a developer machine, as it's a security nightmare. :cold_sweat:

**Reply by Not-Sarthak** [reply]
oh okay got it. thank you! I am looking forward to contribute and have started understanding the codebase today. can you please help me with finding a communications channel where i can ping maintainers to ask doubts as i am still learning a lot.

**Reply by martinpitt** [reply]
On the bottom of https://cockpit-project.org/ you find our mailing list and our Matrix channel. Just be aware that the entire team is on end-of-year holidays, so the channel will be rather deserted until Jan 2. In the meantime, I recommend working through https://github.com/cockpit-project/cockpit/blob/main/HACKING.md to get a development env set up, and ensuring that you can locally build and observe a simple change. Thanks!

**Reply by Not-Sarthak** [reply]
> On the bottom of https://cockpit-project.org/ you find our mailing list and our Matrix channel. Just be aware that the entire team is on end-of-year holidays, so the channel will be rather deserted until Jan 2. In the meantime, I recommend working through https://github.com/cockpit-project/cockpit/blob/main/HACKING.md to get a development env set up, and ensuring that you can locally build and observe a simple change. Thanks!

okay, thanks a lot!
